tilespec.ts

```diff
@@ -1,54 +1,213 @@
-const MAX_TUPLE_MEMBER_VALUE = Number.MAX_SAFE_INTEGER;
-
 /**
  * TileSpec represents a rectangular area on display by means of specifying a
  * number of evenly spaced tiles and two corners.
  */
 export class TileSpec {
     readonly gridWidth: number;
```

Comment:
Is grid width and height no longer required? If so, I would make these number|null.